eldritch/gate_encounters.py

```diff
@@ -20,64 +23,393 @@ def encounter_event(self, character, world_name):
+  return events.Sequence([
+      check,
+      events.Conditional(char, check, "successes", {0: cave, 2: dreamlands, 3: temple}),
+  ], char)
+
+
+def GreatHall4(char) -> events.Event:
+  check = events.Check(char, "fight", -1)
+  spells = events.Draw(char, "spells", 3, keep_count=2)
+  loss = events.Loss(char, {"stamina": 3})
+  return events.PassFail(char, check, loss, spells, min_successes=2)
+
+
+def Other4(char) -> events.Event:
+  return events.MonsterAppears(char)
+
+
+def Dreamlands5(char) -> events.Event:
+  return events.Nothing()
+
+
+def Abyss5(char) -> events.Event:
+  check = events.Check(char, "luck", -1)
+  fishy_object = events.Sequence([
+      events.Gain(char, {"dollars": 3}),
+      events.Draw(char, "unique", 1),
+  ], char)
+  return events.PassFail(char, check, fishy_object, events.Nothing())
+
+
+def Other5(char) -> events.Event:
+  check = events.Check(char, "luck", -1)
+  gain = events.Gain(char, {"clues": values.Calculation(check, "successes")})
+  return events.Sequence([check, gain], char)
+
+
+def Dreamlands6(char) -> events.Event:
+  check = events.Check(char, "speed", -1)
+  loss = events.Loss(char, {"dollars": float("inf")})
+  return events.PassFail(char, check, events.Nothing(), loss)
+
+
+def GreatHall6(char) -> events.Event:
+  check = events.Check(char, "lore", -1)
+  return events.PassFail(char, check, events.Nothing(), events.Curse(char))
+
+
+def Other6(char) -> events.Event:
+  return events.MonsterAppears(char)
+
+
+def Dreamlands7(char) -> events.Event:
+  check = events.Check(char, "luck", -1)
+  clues = events.Gain(char, {"clues": 2})
+  delay = events.Delayed(char)
+  return events.PassFail(char, check, clues, delay)
+
+
+def GreatHall7(char) -> events.Event:
+  check = events.Check(char, "luck", -1)
+  gain = events.Gain(char, {"sanity": 2, "stamina": 2})  # TODO: Source?
+  loss = events.LoseItems(char, 2)
+  return events.PassFail(char, check, gain, loss)
+
+
+def Other7(char) -> events.Event:
+  check = events.Check(char, "fight", -2)
+  gain = events.Gain(char, {"dollars": 8})
+  loss = events.Loss(char, {"stamina": 1})
+  return events.PassFail(char, check, gain, loss)
+
+
+def Dreamlands8(char) -> events.Event:
+  check = events.Check(char, "lore", 0)
+  freeze = events.LostInTimeAndSpace(char)
+  return events.PassFail(char, check, events.Return(char, char.place.info.name), freeze)
+
+
+def Pluto8(char) -> events.Event:
+  check = events.Check(char, "speed", -1)
+  statue = events.Gain(char, {"dollars": 5, "clues": 2})
+  lost = events.LostInTimeAndSpace(char)
+  return events.PassFail(char, check, statue, lost)
+
+
+def Other8(char) -> events.Event:
+  return events.Draw(char, "common", 1)
+
+
+def Dreamlands9(char) -> events.Event:
+  check = events.Check(char, "sneak", 0)
+  lose_item_count = values.Calculation(
+      values.ItemDeckCount(char, {"common", "unique", "spells", "tradables"}), None,
```

Comment:
I guess I interpreted it as `lose {half your items (your choice) and half your money}(rounded up)` rather than `lose {half your items (your choice)} and {half your money (rounded up)}`. Why didn't they write these in Lojban?